mslice/scripting/__init__.py

```diff
@@ -22,15 +21,57 @@ def generate_script(ws_name, filename=None, plot_handler=None, window=None):
 
 def preprocess_lines(ws_name, plot_handler):
     from mslice.plotting.plot_window.cut_plot import CutPlot
+    script_lines = []
 
     if isinstance(plot_handler, CutPlot) and len(get_cut_plotter_presenter()._cut_cache) > 1:
-        script_lines = []
-        for workspace_name in get_cut_plotter_presenter()._cut_cache:
-            ws = get_workspace_handle(workspace_name).raw_ws
-            script_lines += ["\nws_{} = mc.".format(workspace_name.replace(".", "_"))] + cleanup(
-                GeneratePythonScript(ws).split('\n'))
+        cut_cache = get_cut_plotter_presenter()._cut_cache
+        cache_list = get_cut_plotter_presenter()._cut_cache_list
+        for workspace_name in cut_cache:
+            if any([workspace_name.replace(".", "_") == cut.workspace_name for cut in cache_list]):
+                ws = get_workspace_handle(workspace_name).raw_ws
+                script_lines += generate_script_lines(ws, workspace_name)
     else:
         ws = get_workspace_handle(ws_name).raw_ws
-        script_lines = ["\nws = mc."] + cleanup(GeneratePythonScript(ws).split('\n'))
+        script_lines += generate_script_lines(ws, ws_name)
 
     return script_lines
+
+
+def generate_script_lines(raw_ws, workspace_name):
+    lines = []
+    ws_name = workspace_name.replace(".", "_")
+    alg_history = raw_ws.getHistory().getAlgorithmHistories()[::-1]
+    for algorithm in alg_history:
+        alg_name = algorithm.name()
+        kwargs = get_algorithm_kwargs(algorithm, ws_name)
+        if alg_name != 'Load':
+            lines += ["ws_{} = mc.{}({})\n".format(ws_name, alg_name, kwargs)]
+        else:
+            lines += ["ws_{} = mc.{}({})\n".format(ws_name, alg_name, kwargs)]
+            break
+    return lines[::-1]
+
+
+def get_algorithm_kwargs(algorithm, workspace_name):
+    arguments = []
+    if algorithm.name() == "Load":
+        for property in algorithm.getProperties():
+            if property.name() == "Filename":
```

Comment:
I think it might be better to blacklist the properties we don't want, i.e skip `LoaderName`, `OutputWorkspace` and any that are `isDefault` so that any added in the future would not be trapped by this.